plugins/inputs/tengine/tengine.go

```diff
@@ -0,0 +1,335 @@
+package tengine
+
+import (
+	"bufio"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/internal/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+)
+
+type Tengine struct {
+	Urls            []string
+	ResponseTimeout internal.Duration
+	tls.ClientConfig
+
+	// HTTP client
+	client *http.Client
+}
+
+var sampleConfig = `
+  # An array of Nginx stub_status URI to gather stats.
+  urls = ["http://localhost/server_status"]
```

Comment:
I think we should use `"http://127.0.0.1/us"` as the default since it matches the example [here](http://tengine.taobao.org/document/http_reqstat.html) and may help confusion with `ngx_http_stub_status_module`